### ISSUE TITLE
Empty tag: URL

### ISSUE THREAD

**Issue by FrostyX** [fedora-ruby/gem2rpm#71]
Lot of my attempts to use `gem2rpm <gem> --fetch --srpm` ends with following error:

    $ gem2rpm a1447ll_mini_test --fetch --srpm
    error: line 10: Empty tag: URL:

I can see that such gem has empty `homepage:` in its specification and therefore URL in generated .spec file is also empty, but

would it be possible to set default URL to `https://rubygems.org/gems/<GEM>` for such gems? And maybe print warning, but not end up with fatal error. Maybe it could be turned on/off by some optional parameter. There is nothing to lose.

What do you think?

PS: I've tried to build first (alphabetically) 1000 gems from RubyGems and 67 of them failed just because of this error.

**Comment by junaruga** [comment]
It looks useful feature.

As a information, above error might happen on rpmbuild command in bin/gem2rpm.
https://github.com/fedora-ruby/gem2rpm/blob/v0.11.3/bin/gem2rpm#L71

Its error has not handled right now.
Below my pull-request will fix the error handle.
https://github.com/fedora-ruby/gem2rpm/pull/64

**Comment by voxik** [comment]
@FrostyX Just to make sure, on what system are you testing? I am asking since this is the content of default Fedora 21 - Rawhide template:

```
<% if spec.homepage -%>
URL: <%= spec.homepage %>
<% end -%>
```

but the default template, which is used when system is not recognized, does not contain the condition.

But the intention definitely was to not emit the URL if it is not set, but it might be actually empty string .....

Also, you might still consider to create your own template, which will not contain the URL at all ...

**Comment by FrostyX** [comment]
It was tested on Fedora 23

**Comment by voxik** [comment]
@FrostyX thx. Then the spec.homepage probably returns empty string and we should normalize it, e.g. using something as #blank?

**Comment by FrostyX** [comment]
Thank you @voxik for suggestions.

> using something as #blank?

Do you mean generating something like:

    License: MIT
    URL: #blank
    Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem

? Honestly I would rather see URL to gem on rubygems.org, but it is up to you. For me it is even `#blank` good enough, because at least it will proceed to building the srpm file. Therefore I have no problem with such solution.

It is also ok for me to not generate the URL line at all.

**Comment by voxik** [comment]
@FrostyX No, don't worry ... I was thinking about fixing the issue ... blank? returns true for nil and empty values

**Comment by junaruga** [comment]
I will fix this!

**Comment by junaruga** [comment]
Let me do it!

**Comment by junaruga** [comment]
Fixed. Please check my pull-request.

**Comment by junaruga** [comment]
I updated with another commit.
Could you check again?
If my modification is ok for you, could you tell me?
 I can do squash to unify my commits to 1 commit.

**Comment by junaruga** [comment]
My modification may be merged to master branch soon.

I checked below command was ok with my modification in my environment.
```
$ bin/gem2rpm a1447ll_mini_test --fetch --srpm
Wrote: /home/jaruga/git/gem2rpm_junaruga/rubygem-a1447ll_mini_test-0.1.0-1.fc23.src.rpm
```

**Comment by voxik** [comment]
Should be fixed by 353b2b26572301a1dedd143f90ae75a12e731e19

**Comment by FrostyX** [comment]
Thank you guys for fixing this.
I just tried gem2rpm from master and it is great. See results of following experiment.

I have submitted same 1000 gems to copr and first used `0.11.3` version and then `master`.

`0.11.3`: 474 succeeded
`master`: 548 succeeded
